src/core/src/buffer.rs

```diff
@@ -103,6 +103,8 @@ pub enum Role {
     Index,
     /// Constant buffer
     Constant,
+    /// Staging buffer
+    Staging,
```

Comment:
I think you are right. Staging buffers don't fulfil any role directly (other than being staging, obviously), so your initial approach is correct. For GL, I suspect the staging buffer will really be a PBO target.